packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -323,25 +353,29 @@ export class DocumentDeltaConnection
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
         this.socket.on("signal", this.earlySignalHandler!);
 
+        let success = false;
+
         this._details = await new Promise<IConnected>((resolve, reject) => {
+            const fail = (socketProtocolError: boolean, err: DriverError) => {
```

Comment:
Should this also set `this.closed = true`?

Reply:
Yes, I think it makes sense, though it's a bit funny, as we are getting more and more code duplication between close() and fail(0 methods :)